tests/integration/test_target_branch_protection.py

```diff
@@ -0,0 +1,232 @@
+):
+    """
+    arrange: given a branch that is not compliant.
+    act: when target_branch_protection is called with the name of the branch.
+    assert: then a fail report is returned.
+    """
+    if branch_protection_parameters.branch_protection_enabled:
+        edit_branch_protection(github_branch, branch_protection_parameters)
+
+    # The github_client is injected
+    report = target_branch_protection(  # pylint: disable=no-value-for-parameter
+        repository_name=github_repository_name, branch_name=github_branch.name
+    )
+
+    if branch_protection_parameters.branch_protection_enabled:
+        github_branch.remove_protection()
+
+    assert_substrings_in_string(reason_string_array, str(report.reason))
+    assert github_branch.name in str(report.reason)
+    assert report.result == Result.FAIL
+
+
+@pytest.mark.parametrize("github_branch", [f"protected/{uuid4()}"], indirect=True)
+def test_pass(
+    github_branch: Branch,
+    github_repository_name: str,
+):
+    """
+    arrange: given a branch that is compliant.
+    act: when target_branch_protection is called with the name of the branch.
+    assert: then a pass report is returned.
+    """
+    branch_protection_parameters = BranchProtectionParameters(
+        branch_protection_enabled=True,
+        require_code_owner_reviews=True,
+        dismiss_stale_reviews_enabled=True,
+        bypass_pull_request_allowance_disabled=True,
+        required_signatures_enabled=True,
+    )
+    edit_branch_protection(github_branch, branch_protection_parameters)
+
+    # The github_client is injected
+    report = target_branch_protection(  # pylint: disable=no-value-for-parameter
+        repository_name=github_repository_name, branch_name=github_branch.name
+    )
+    github_branch.remove_protection()
+
+    assert report.reason is None
+    assert report.result == Result.PASS
+
+
+def test_no_github_token(github_repository_name: str, monkeypatch: pytest.MonkeyPatch):
+    """
+    arrange: A github repository name and a missing github token.
+    act: when the github client is injected to target_branch_protection.
+    assert: An InputError is raised with a specific error message.
+    """
+    monkeypatch.delenv(GITHUB_TOKEN_ENV_NAME)
+    with pytest.raises(InputError) as exc:
+        # The github_client is injected
+        target_branch_protection(  # pylint: disable=no-value-for-parameter
+            repository_name=github_repository_name, branch_name="arbitrary"
+        )
+        assert_substrings_in_string([GITHUB_TOKEN_ENV_NAME, "was not provided"], str(exc))
+
+
+def edit_branch_protection(
+    github_branch: Branch, branch_protection_parameters: BranchProtectionParameters
+) -> None:
+    """Enable or disable the branch protection parameters on the given branch.
+
+    Args:
+        github_branch: The branch to modify.
+        branch_protection_parameters: The named tuple giving the list of parameters
+        to enable or disable.
```

Comment:
Should this be indented by one more level?

Reply:
Not relevant anymore following some refactoring